tests/unit/config/test_configfiles.py

```diff
@@ -135,6 +135,135 @@ def test_yaml_oserror(fake_save_manager, config_tmpdir):
+        assert config.instance._values['bindings.commands'] == expected
+
+    def test_clear_path(self, confpy, qbmodulepy, tmpdir):
+        qbmodulepy.write("""def run(config):
+    config.bind(",a", "message-info foo", mode="normal")""")
+        confpy.write("""import qbmodule
+qbmodule.run(config)""")
+        api = configfiles.read_config_py(confpy.filename)
+        assert len(api.errors) == 0
+        assert tmpdir not in sys.path
+
+    def test_clear_modules(self, confpy, qbmodulepy):
+        qbmodulepy.write("""def run(config):
+    config.bind(",a", "message-info foo", mode="normal")""")
+        confpy.write("""import qbmodule
+qbmodule.run(config)""")
+        api = configfiles.read_config_py(confpy.filename)
+        assert len(api.errors) == 0
+        assert "qbmodule" not in sys.modules.keys()
+
+    def test_clear_modules_on_err(self, confpy, qbmodulepy):
+        qbmodulepy.write("""def run(config):
+    1/0""")
+        confpy.write("""import qbmodule
+qbmodule.run(config)""")
+        api = configfiles.read_config_py(confpy.filename)
+
+        assert len(api.errors) == 1
+        error = api.errors[0]
+        assert error.text == "Unhandled exception"
+        assert isinstance(error.exception, ZeroDivisionError)
+
+        tblines = error.traceback.strip().splitlines()
+        assert tblines[0] == "Traceback (most recent call last):"
+        assert tblines[-1] == "ZeroDivisionError: division by zero"
+        assert "    1/0" in tblines
+        assert "qbmodule" not in sys.modules.keys()
+
+    def test_clear_path_on_err(self, confpy, qbmodulepy, tmpdir):
+        qbmodulepy.write("""def run(config):
+    1/0""")
+        confpy.write("""import qbmodule
+qbmodule.run(config)""")
+        api = configfiles.read_config_py(confpy.filename)
+
+        assert len(api.errors) == 1
+        error = api.errors[0]
+        assert error.text == "Unhandled exception"
+        assert isinstance(error.exception, ZeroDivisionError)
+
+        tblines = error.traceback.strip().splitlines()
+        assert tblines[0] == "Traceback (most recent call last):"
+        assert tblines[-1] == "ZeroDivisionError: division by zero"
+        assert "    1/0" in tblines
+        assert tmpdir not in sys.path
+
+    def test_fail_on_nonexistent_module(self, confpy, qbmodulepy, tmpdir):
+        qbmodulepy.write("""def run(config):
+    pass""")
+        confpy.write("""import foobar
+foobar.run(config)""")
+        api = configfiles.read_config_py(confpy.filename)
+
+        assert len(api.errors) == 1
+        error = api.errors[0]
+        assert error.text == "Unhandled exception"
+        assert isinstance(error.exception, ImportError)
+
+        tblines = error.traceback.strip().splitlines()
+        assert tblines[0] == "Traceback (most recent call last):"
+        assert (tblines[-1] == "ImportError: No module named 'foobar'" or
+                tblines[-1] == "ModuleNotFoundError: No module named 'foobar'")
```

Comment:
Since python changed it's importerror syntax, something like this is needed, but feels hacky. Is there a better way to do this? Maybe I should leave this check off and leave it at the `assert isinstance(error.exception, ImportError)`

Reply:
Or maybe just `assert tblines[-1].endswith("Error: No module named 'foobar'")` or so?

(haven't looked at the rest yet, busy with other stuff still)